### PR TITLE
Communication interface class

### PR DESCRIPTION
**Pull request by maxmykhailenko**
I have some open question which would like to ask. After that I can add the class description.

### CODE REVIEW

**Comment by pcingola** [comment]
I think we need a method for feature importances that optionally returns the importances from the model if they exist (or None if they don't).

File: communication_interface.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    """
+    This class describing the structure of Model which could be integrated in LogML project.
+    A model that can be created, trained and evaluated.
+    Inherited from Configurable class which set all parameters from config to class parameters.
+    """
+
+    def __init__(self, config, config_section=None, datasets=None, set_config=True):
+        self.config_section = ('feature_importance', 'model_search') if config_section is None else config_section
+        super().__init__(congig=config, config_section=config_section)
+
```
Comment (pcingola) [inline communication_interface.py:15]: Typo `congig=config`

File: communication_interface.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    def predict(self, *args, **kwargs):
+        """
+        After a model has been fit predict returns the fitted values.
+        This is a placeholder intended to be overwritten by individual models.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def save(file_pickle, tag, data):
+        """ Save dataset to pickle file """
```
Comment (pcingola) [inline communication_interface.py:29]: Why do you need `data` to save? Isn't this supposed to save the Model object (i.e. `self`)?

Comment (pcingola) [reply communication_interface.py:29]: What is `tag` for? Why are you forcing all models to pass this parameter to save the model?

File: communication_interface.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            logger.debug(f"{tag}: Empty file name, skipping")
+            return False
+        logger.debug(f"{tag}: Saving pickle file '{file_pickle}'")
+        with open(file_pickle, 'wb') as output:
+            pickle.dump(data, output)
+            return True
+
+    @staticmethod
+    def load(file_pickle: str, tag, config: Config):
+        """Load a pickle file, return data (on success) or None (on failure)"""
```
Comment (pcingola) [inline communication_interface.py:40]: Why do you need `tag` here and why is it a mandatory parameter?

File: communication_interface.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        if not file_pickle:
+            logger.debug(f"{tag}: Empty file name. Get file name from config file")
+            file_pickle = config.get_file_pickle()
+            if not file_pickle:
```
Comment (pcingola) [inline communication_interface.py:44]: If you only require the `config` file when `file_pickle` is no passed, shouldn't `file_pickle` and `config` have default None values?

**Review by pcingola — COMMENTED** [review]
Each method MUST be thoroughly documented.

File: communication_interface.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        This is a placeholder intended to be overwritten by individual models.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def save(file_pickle, tag, data):
+        """ Save dataset to pickle file """
+        if not file_pickle:
+            logger.debug(f"{tag}: Empty file name, skipping")
+            return False
```
Comment (pcingola) [inline communication_interface.py:32]: What is `logger`? Where is this coming from?

File: communication_interface.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            return dataset
+
+    def get_feature_importances(self):
+        """
+        Return importance from the model if exist, else return None if they don't.
+        Calculate the average of all importances if cross-validation enabled.
+        :return:
+        """
+        if not self.cv_enable:
+            return self.model.feature_importances_
```
Comment (pcingola) [inline communication_interface.py:61]: Where is `self.cv_enable` defined?
Where are these `self.model` and `self.model.feature_importances_` comming from?

**Review by pcingola — CHANGES_REQUESTED** [review]

**Comment by nchaly** [comment]
Closing this as outdated request.

**Pull request closed without merging**